src/lurch.c

```diff
@@ -1599,8 +1600,8 @@ static void lurch_message_encrypt_groupchat(PurpleConnection * gc_p, xmlnode **
     // and the typed message written to the chat window manually without sending
     if (!g_strcmp0(curr_muc_member_jid, uname)) {
       body_node_p = xmlnode_get_child(*msg_stanza_pp, "body");
-
-      purple_conv_chat_write(chat_p, curr_muc_member_p->handle, xmlnode_get_data(body_node_p), PURPLE_MESSAGE_SEND, time((void *) 0));
+      body_data = xmlnode_get_data(body_node_p);
+      purple_conv_chat_write(chat_p, curr_muc_member_p->handle, body_data, PURPLE_MESSAGE_SEND, time((void *) 0));
       continue;
```

Comment:
Doesn't the `g_free()` need to be before the continue?

Reply:
Please review again. Thanks.